quinn-proto/src/endpoint.rs

```diff
@@ -618,15 +617,7 @@ impl Endpoint {
                 debug!("handshake failed: {}", e);
                 self.handle_event(ch, EndpointEvent(EndpointEventInner::Drained));
                 if let ConnectionError::TransportError(e) = e {
-                    self.initial_close(
-                        version,
-                        remote,
-                        local_ip,
-                        crypto,
-                        &src_cid,
-                        &temp_loc_cid,
-                        e,
-                    );
+                    self.initial_close(version, remote, local_ip, crypto, &src_cid, &loc_cid, e);
```

Comment:
In this case we end up with a connection with the `loc_cid` but we also send an initial close with that `loc_cid`. That might not be what we want?

Reply:
The `handle_event` call above cleans up the connection. Perhaps this could be refactored not to add it in the first place, but the existing behavior looks correct to me.